packages/devtools_app/lib/src/connected_app.dart

```diff
@@ -8,36 +8,66 @@ import 'globals.dart';
 import 'service_extensions.dart' as extensions;
 
 const flutterLibraryUri = 'package:flutter/src/widgets/binding.dart';
-const flutterWebLibraryUri = 'package:flutter_web/src/widgets/binding.dart';
 const dartHtmlLibraryUri = 'dart:html';
 
 class ConnectedApp {
   ConnectedApp();
 
-  Future<bool> get isFlutterApp async =>
-      _isFlutterApp ??= await _libraryUriAvailable(flutterLibraryUri);
+  bool get appTypeKnown => _appTypeKnown;
+  bool _appTypeKnown = false;
+
+  FutureOr<bool> get isFlutterApp async {
+    _isFlutterApp ??= await _libraryUriAvailable(flutterLibraryUri);
+    _updateAppTypeKnown();
+    return _isFlutterApp;
+  }
+
+  bool get isFlutterAppRaw {
+    assert(_isFlutterApp != null);
+    return _isFlutterApp;
+  }
+
   bool _isFlutterApp;
 
-  Future<bool> get isPackageFlutterWeb async =>
-      _isPackageFlutterWeb ??= await _libraryUriAvailable(flutterWebLibraryUri);
-  bool _isPackageFlutterWeb;
+  FutureOr<bool> get isProfileBuild async {
+    _isProfileBuild ??= await _connectedToProfileBuild();
+    _updateAppTypeKnown();
+    return _isProfileBuild;
+  }
+
+  bool get isProfileBuildRaw {
+    assert(_isProfileBuild != null);
+    return _isProfileBuild;
+  }
 
-  Future<bool> get isProfileBuild async =>
-      _isProfileBuild ??= await _connectedToProfileBuild();
   bool _isProfileBuild;
 
-  Future<bool> get isAnyFlutterApp async =>
-      await isFlutterApp || await isPackageFlutterWeb;
+  FutureOr<bool> get isDartWebApp async {
+    _isDartWebApp ??= await _libraryUriAvailable(dartHtmlLibraryUri);
+    _updateAppTypeKnown();
+    return _isDartWebApp;
+  }
+
+  bool get isDartWebAppRaw {
+    assert(_isDartWebApp != null);
+    return _isDartWebApp;
+  }
 
-  Future<bool> get isDartWebApp async =>
-      _isDartWebApp ??= await _libraryUriAvailable(dartHtmlLibraryUri);
   bool _isDartWebApp;
 
+  void _updateAppTypeKnown() {
+    _appTypeKnown = _isFlutterApp != null &&
+        _isProfileBuild != null &&
+        _isDartWebApp != null;
+  }
+
   bool get isRunningOnDartVM => serviceManager.vm.name != 'ChromeDebugProxy';
 
-  Future<bool> get isDartCliApp async =>
+  FutureOr<bool> get isDartCliApp async =>
```

Comment:
I think `Future<bool>` is correct here.

Reply:
since isFlutterApp returns a FutureOr, wouldn't it be FutureOr?

Reply:
I think they can both be changed to `Future`?

It's not a critical change.